sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/QueryWrapperGenerator.kt

```diff
@@ -45,20 +48,32 @@ import com.squareup.sqldelight.core.lang.util.rawSqlText
 
 internal class QueryWrapperGenerator(module: Module, sourceFile: SqlDelightFile) {
   private val sourceFolders = SqlDelightFileIndex.getInstance(module).sourceFolders(sourceFile)
+  private val moduleFolders = SqlDelightFileIndex.getInstance(module)
+      .sourceFolders(sourceFile, includeDependencies = false)
   private val fileIndex = SqlDelightFileIndex.getInstance(module)
 
-  fun type(): TypeSpec {
+  fun interfaceType(): TypeSpec {
+    val typeSpec = TypeSpec.interfaceBuilder(fileIndex.className)
+        .addSuperinterface(TRANSACTER_TYPE)
 
-    val typeSpec = TypeSpec.classBuilder(fileIndex.className)
-        .superclass(TRANSACTER_TYPE)
-        .addSuperclassConstructorParameter(DRIVER_NAME)
+    fileIndex.dependencies.forEach { (packageName, className) ->
+      typeSpec.addSuperinterface(ClassName(packageName, className))
+    }
 
-    val constructor = FunSpec.constructorBuilder()
+    val implType = ClassName(fileIndex.packageName, "${fileIndex.className}Impl")
+
+    val invoke = FunSpec.builder("invoke")
```

Comment:
We should probably have `@JvmName("create") @JsName("create")` and whatever the native equivalent is so it's idiomatic for non-Kotlin callers.

Reply:
And `@JvmStatic`

Reply:
ill add these later if at all. The pattern ive been doing now is writing my own method which also provides adapters in the kotlin codebase.